### DISCUSSION TITLE
Mouse VDJ Data

### DISCUSSION THREAD

**Discussion by hungms** [tuonglab/dandelion#310] [General]
Hi Kelvin,

Thanks for developing this great package. 

I am currently analysing **mouse** VDJ data with your package in the singularity container but I see all my IGHA calls from igblast were assigned as IGHA2 after correction. 

[2MpOG_1A.pdf](https://github.com/zktuong/dandelion/files/12521368/2MpOG_1A.pdf)

I see that in **Step 4: Assigning constant region calls**, you mentioned: 
> Further, to deal with incorrect constant gene calls due to insufficient length, a pairwise alignment will be run against [curated sequences](https://immunology.sciencemag.org/content/6/56/eabe6291) that were deemed to be highly specific in distinguishing IGHA1 vs IGHA2, and IGHG1 to IGHG4.

Can I check if the IGHA2 calls coming from this step (the curated sequences are from human I presume?). If this is the case, can I presume that it won't be appropriate to run this step for mouse VDJ to correct the other constant region calls? Or is there an alternative approach to this?

Thank you again in advance.

**Comment by zktuong** [comment]
hi @hungms, yes you are spot on. 

You can run `assign_isotype` with `correct_c_call=False` to disable the correction step - that should alleviate the issue.

otherwise, provide a dictionary to `correction_dict` argument - here's a dictionary for mouse IGHG:

```python
"""Primer sequences for mouse ighg"""
    primer_dict = {
        "IGHG2": {
            "IGHG2A": "GCCAAAACAACAGCCCCATCGGTCTATCCACTGGCCCCTGTGTGTGGAGATACAACTGGC",
            "IGHG2B": "GCCAAAACAACACCCCCATCAGTCTATCCACTGGCCCCTGGGTGTGGAGATACAACTGGT",
            "IGHG2C": "GCCAAAACAACAGCCCCATCGGTCTATCCACTGGCCCCTGTGTGTGGAGGTACAACTGGC",
        }
    }
```
